src/Text/Layout/Table/Cell.hs

```diff
@@ -72,6 +119,32 @@ instance Cell T.Text where
 
     buildCell = textB
 
+-- | An object with padding on the left and right.
+data Padded a
+    = Padded
+    { paddedObject :: a
+    , leftPadding  :: Int
+    , rightPadding :: Int
+    } deriving (Eq, Ord, Show, Functor)
+
+instance Applicative Padded where
+  pure a = Padded a 0 0
+  (Padded f l r) <*> (Padded a l' r') = Padded (f a) (l + l') (r + r')
+
+instance Monad Padded where
+  (Padded a l r) >>= f = let Padded b l' r' = f a in Padded b (l + l') (r + r')
+
+-- | The total amount of padding in 'Padded'.
+totalPadding :: Padded a -> Int
+totalPadding a = leftPadding a + rightPadding a
+
+-- | Redistribute the padding using a given ratio.
+redistributePadding :: Int -> Int -> Padded a -> Padded a
+redistributePadding l r padded = Padded (paddedObject padded) lPadding rPadding
+  where
+    lPadding = (totalPadding padded * l) `div` (l + r)
+    rPadding = totalPadding padded - lPadding
```

Comment:
I guess since this change is already in it does no longer matter.  But it would still be interesting to know what happens with negative values.

Reply:
It would redistribute them in the same way: so if you were going to trim two from the left and one from the right, it could redistribute that to 1 from the left and two from the right, for example.

Reply:
Right.  Although, `l + r == 0` and `signum l /= signum r` are both a bit weird.  But I guess we can assume that only reasonable input is passed.

Reply:
Yes, I think that's safe. I'm actually not sure this function will be needed anymore. This PR is being redone in light of the new lazier Cell type, and it might disappear.